### PR TITLE
Truncate long pool names and fallback to id if no name exists

### PR DESCRIPTION
**Pull request by kimpers**
40 characters is very arbitrary and we can change that but I think it looks ok with that length.

image

image

### CODE REVIEW

File: ts/components/staking/wizard/wizard_flow.tsx [hunk 1]
```diff
@@ -638,7 +638,7 @@ const StartStaking: React.FC<StartStakingProps> = props => {
                               `You're delegating ${stakingAmountTotalComputed} ZRX to ${
                                   selectedStakingPools.length > 1
                                       ? `${selectedStakingPools.length} pools`
-                                      : `${selectedStakingPools[0].pool.metaData.name || '1 pool'}`
+                                      : `${utils.getPoolDisplayName(selectedStakingPools[0].pool)}`
```
Comment (kimpers) [inline ts/components/staking/wizard/wizard_flow.tsx]: Here I was a bit lazy but I think `Pool 1` works as well but open to changing it into a custom truncation here

File: ts/components/staking/wizard/wizard_flow.tsx [hunk 2]
```diff
@@ -432,7 +432,7 @@ const MarketMakerStakeInputPane = (props: MarketMakerStakeInputPaneProps) => {
             <PoolsContainer>
                 <MarketMaker
                     key={marketMakerPool.poolId}
-                    name={marketMakerPool.metaData.name || utils.getAddressBeginAndEnd(marketMakerPool.operatorAddress)}
```
Comment (kimpers) [inline ts/components/staking/wizard/wizard_flow.tsx:435]: We were a bit inconsistent using both `poolId` and `operatorAddress` as a fallback if no name exists so I standardized it to use `poolId`.

File: ts/utils/utils.ts [hunk 3]
```diff
@@ -556,4 +556,17 @@ export const utils = {
 
         return `0x${_n.toString(16).padStart(64, '0')}`;
     },
+    getPoolDisplayName({ poolId, metaData }: { poolId: string; metaData?: { name?: string } }): string {
```
Comment (johnrjj) [inline ts/utils/utils.ts]: nit: (no preference either way) would this better belong in staking_utils instead?

**Review by johnrjj — APPROVED** [review]
LGTM -- thanks for doing this!

**Pull request merged**